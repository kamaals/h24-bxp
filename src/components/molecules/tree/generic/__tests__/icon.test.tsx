import { describe, it, expect, beforeEach } from "vitest";
import { render, screen, cleanup } from "@testing-library/react";
import Icon from "../icon";
import { Eye, ShieldEllipsis } from "lucide-react";

const data = {
  id: "1",
  name: "Test",
  icon: Eye,
};

describe(Icon.name, () => {
  beforeEach(() => {
    cleanup();
  });

  it("Should render correctly for selected", () => {
    render(<Icon data={data} />);
    expect(screen.getByTestId("icon")).toBeInTheDocument();
  });

  it("Should render correctly for selected", () => {
    render(
      <Icon
        data={{ ...data, selectedIcon: ShieldEllipsis }}
        isSelected={true}
      />,
    );
    expect(screen.getByTestId("selected-icon")).toBeInTheDocument();
  });

  it("Should render correctly for opened", () => {
    render(<Icon data={{ ...data, openIcon: ShieldEllipsis }} isOpen={true} />);
    expect(screen.getByTestId("open-icon")).toBeInTheDocument();
  });

  it("Should match the snapshot", () => {
    const resp = render(<Icon data={data} />);
    expect(resp).toMatchSnapshot();
  });
});
