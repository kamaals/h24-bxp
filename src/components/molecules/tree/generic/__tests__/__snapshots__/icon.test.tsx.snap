// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TreeIcon > Should match the snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span>
        <svg
          aria-hidden="true"
          class="lucide lucide-eye h-4 w-4 shrink-0 mr-2 relative z-20"
          data-testid="icon"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M2.062 12.348a1 1 0 0 1 0-.696 10.75 10.75 0 0 1 19.876 0 1 1 0 0 1 0 .696 10.75 10.75 0 0 1-19.876 0"
          />
          <circle
            cx="12"
            cy="12"
            r="3"
          />
        </svg>
      </span>
    </div>
  </body>,
  "container": <div>
    <span>
      <svg
        aria-hidden="true"
        class="lucide lucide-eye h-4 w-4 shrink-0 mr-2 relative z-20"
        data-testid="icon"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M2.062 12.348a1 1 0 0 1 0-.696 10.75 10.75 0 0 1 19.876 0 1 1 0 0 1 0 .696 10.75 10.75 0 0 1-19.876 0"
        />
        <circle
          cx="12"
          cy="12"
          r="3"
        />
      </svg>
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
